Lesson_2/Exercise_1/BankAccount.cs

```diff
@@ -1,82 +1,152 @@
 using System.Text;
 using System.Threading.Tasks;
 
 namespace Exercise_1
 {
-    enum TypeOfAccount { checkingAcnt, savingsAcnt, investmentAcnt };           // все типы банковских счетов
-    class BankAccount
-    {       
-        //private long accountNumber;                                              // номер счета
+    
+    public enum TypeOfAccount { CheckingAcnt, SavingsAcnt, InvestmentAcnt };           // все типы банковских счетов
+    public class BankAccount
+    {                                                    
         private double balance;                                                 // баланс счета
         private TypeOfAccount typeOfAcnt;                                       // тип банковского счета
+        private long acntNum;                                                   // номер счета
         private static long unicAcntNum;                                        // уникальный номер счета
+ 
+        public double Balance => this.balance;
+       
+        public TypeOfAccount TypeOfAccount => this.typeOfAcnt;
+       
+        public long UnivAcntNum => unicAcntNum;
 
-        public double Balance
-        {
-            get => this.balance; 
-            //set => this.balance = value;
-        }
-        public TypeOfAccount TypeOfAccount
+        public long AcntNum
         {
-            get => this.typeOfAcnt; 
-            set => this.typeOfAcnt = value;
+            get => this.acntNum;
         }
-        public long UnivAcntNum => unicAcntNum;
-
-        //public BankAccount(double balance)
-        //{
-        //    this.balance = balance;
-        //    GenNewAcntNum();
-        //}
-
-        //public BankAccount(TypeOfAccount typeOfAcnt)
-        //{
-        //    this.typeOfAcnt = typeOfAcnt;
-        //    GenNewAcntNum();
-        //}
-
+        
         public BankAccount(double balance, TypeOfAccount typeOfAcnt)
         {
             this.balance = balance;
             this.typeOfAcnt = typeOfAcnt;
             GenNewAcntNum();
+            acntNum = unicAcntNum;
         }
-
-        public static long GenNewAcntNum()                                      // генерирование нового уникального номера счета
+        /// <summary>
+        /// генерирование нового уникального номера счета
+        /// </summary>
+        /// <returns>уникальный номер счета</returns>
+        public static long GenNewAcntNum()
         {
             return ++unicAcntNum;
         }
-        public void GiveMeMyMoney(double Sum)                                             // заполнение данных счета
+        /// <summary>
+        /// перевод средств со счета
+        /// </summary>
+        /// <param name="Sum"></param>
+        public void GiveMeMyMoney(double Sum)
         {
             if (Sum > balance)
-                Console.WriteLine("Недостаточно средств на счету для данной операции");
+                throw new NotImplementedException($"недостаточно средств на счете: {this.acntNum}");
```

Comment:
```suggestion
                throw new InvalidOperationException($"недостаточно средств на счете: {this.acntNum}");
```
Не вводите в заблуждение! Это не "не реализовано", а "ошибка в операции".